### PR TITLE
DOC: 

### PR DESCRIPTION
**Pull request by TomDonoghue**
I was poking around the codebase, checking through it a bit, and here are a bunch of small fixes. 

Almost entirely doc fixes:
- Fix some spacing issues
- Get better at line length (even if we haven't yet picked a standard, some of it was a bit excessive)
- Small clean ups on var names, type checking, etc. 
- Docstring updates, to note when parameters are optional

### CODE REVIEW

**Comment by TomDonoghue** [comment]
Well, Travis seems to be having a bad day. It's failing to build, I re-launched with a new commit, but still no luck. 

I don't really know what's up. Our CI seems to be a bit unstable, but unclear if it's something on our end, it's just not building properly. A problem for later...

If you want to review it, I barely touched any actual code, and so it's pretty unlikely I actually broke anything. I'll try and re-launch Travis tomorrow and see if it's feeling any better.

**Comment by srcole** [comment]
nice improvements!

oh I didn't realize it was customary to put the "optional" descriptor in docstrings. A bit overkill if you ask me, but I'm fine either way. But it doesn't look like you updated all docstrings, right? so i guess it'll be a bit inconsistent.

yeah, I'll look into travis and see if I can get any insight onto what's going on

**Comment by TomDonoghue** [comment]
@srcole 

Yeah, marking 'optional', is part of numpy docs, perhaps not the most useful part, but at least you can get a sense of what's required without bouncing back & forth from docstring to inputs. 
I can't promise it's entirely inconsistent - but some of the other files did already have it, so it was already inconsistent, and so it should now be less inconsistent, in the right direction.

**Comment by srcole** [comment]
CI is working fine on my PR #33 ... idk what's going on here

File: neurodsp/pac.py [hunk 1]
```diff
@@ -291,6 +291,7 @@ def compute_pac_comodulogram(x_pha, x_amp, Fs,
         phase-amplitude coupling strength values for each combination of phase-providing
         frequency bin and amplitude-providing frequency bin.
     """
+    #TODO: N_cycles_amp is not used. Figure this out.
```
Comment (srcole) [inline neurodsp/pac.py]: Reminder to get rid of this in PR #33

**Comment by srcole** [comment]
looks good to me. @rdgao how are the changes to spectral.py?

File: neurodsp/pac.py (hunk 1 above)
Comment (srcole) [reply neurodsp/pac.py]: nvm. See Tom's committed to get rid of it here

**Comment by TomDonoghue** [comment]
I removed the TODO. 

Looks like Travis was just having a bad day yesterday - seems to be running fine now.

**Comment by srcole** [comment]
I was wondering if it was just that you moved the warnings import

**Comment by TomDonoghue** [comment]
@srcole : I was only reordering the imports (to please lintr) as a trivial thing to push a new commit -> relaunch travis. I really doubt that actually changed or fixed anything meaningful.

**Comment by rdgao** [comment]
spectral.py (and everything else) looks great! definitely a lot more profresh now

**Pull request merged**